django/urls/resolvers.py

```diff
@@ -71,6 +72,27 @@ def __repr__(self):
             )
         )
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        if self.is_class_based:
+            # The following attributes of ResolverMatch prevent pickling.
+            del state['func']
+            del state['tried']
+
+        return state
+
+    def __setstate__(self, state):
+        from django.urls import resolve
+        self.__dict__.update(state)
+
+        if self.is_class_based:
+            # Set the attributes again which could not be pickled.
+            path = '/{}'.format(self.route)
+            match = resolve(path)
+            self.func = match.func
+            self.tried = match.tried
```

Comment:
URL patterns may change over time, so we can restore the different match :thinking: